utils/firestore.js

```diff
@@ -2400,77 +2400,95 @@ const confirmShipmentKit = async (shipmentData) => {
 
 const storeKitReceipt = async (package) => {
     try {
-        const kitSnapshot = await db.collection("kitAssembly").where('972453354', '==', package['972453354']).where('221592017', '==', 277438316).get();
-        if (kitSnapshot.size === 0) {
-            return false;
-        }
-        const kitDoc = kitSnapshot.docs[0];
-        const Connect_ID = kitDoc.data()['Connect_ID'];
-   
-        const participantSnapshot = await db.collection("participants").where('173836415.266600170.319972665.687158491', '==', kitDoc.data()[687158491]).get();
-        const participantDoc = participantSnapshot.docs[0];
-        const participantDocData = participantSnapshot.docs[0].data();
+        let toReturn;
+        await db.runTransaction(async (transaction) => {
+            const kitSnapshot = await transaction.get(db.collection("kitAssembly").where('972453354', '==', package['972453354']).where('221592017', '==', 277438316));
+            if (kitSnapshot.size === 0) {
+                toReturn = false;
+                return;
+            }
+            const kitDoc = kitSnapshot.docs[0];
+            const kitData = kitDoc.data();
+            const Connect_ID = kitData['Connect_ID'];
+    
+            const participantSnapshot = await transaction.get(db.collection("participants").where('173836415.266600170.319972665.687158491', '==', kitDoc.data()[687158491]));
+            const participantDoc = participantSnapshot.docs[0];
+            const participantDocData = participantSnapshot.docs[0].data();
+
+            const token = participantDocData['token'];
+            const uid = participantDocData['state']['uid'];
+            const site = participantDocData['827220437'];
+            const prefEmail = participantDocData['869588347'];
+            const ptName = participantDocData['153211406'] || participantDocData['399159511']
+            const surveyStatus = participantDocData['547363263']
+
+            const prevParticipantObject = participantDocData[173836415][266600170][319972665];
+            const collectionId = package['259846815']?.split(' ')[0];
+            const objectId = package['259846815']?.split(' ')[1];
+            
+            if (objectId === undefined || collectionId === undefined) {
+                toReturn = { status: 'Check Collection ID' };
```

Comment:
Do we need assign value to `toReturn`?

Reply:
In the original, it was `return { status: 'Check Collection ID' }` here, so I'm doing my best to mirror that exact setup.